mermithid/misc/FakeTritiumDataFunctions.py

```diff
@@ -0,0 +1,655 @@
+'''
+Generate binned or pseudo unbinned data
+Author: T. Weiss, C. Claessens, Y. Sun
+Date:4/6/2020
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+import math
+
+from scipy.special import gamma
+from scipy import integrate
+from scipy import stats
+from scipy.optimize import fsolve
+from scipy import constants
+from scipy.interpolate import interp1d
+from scipy.signal import convolve
+import random
+import os
+
+from morpho.utilities import morphologging
+logger = morphologging.getLogger(__name__)
+
+"""
+Constants and functions used by processors/TritiumSpectrum/FakeDataGenerator.py
+"""
+
+"""
+Physical constants
+"""
+
+me = 510999. #eV
+alpha = 1/137.036
+c = 299792458. #m/s
+hbar = 6.58212*10**(-16) #eV*s
+gv=1. #Vector coupling constant
+lambdat = 1.2724 # +/- 0.0023, from PDG (2018): http://pdg.lbl.gov/2018/listings/rpp2018-list-n.pdf
+ga=gv*(-lambdat) #Axial vector coupling constant
+Mnuc2 = gv**2 + 3*ga**2 #Nuclear matrix element
+GF =  1.1663787*10**(-23) #Gf/(hc)^3, in eV^(-2)
+Vud = 0.97425 #CKM element
+
+#Beta decay-specific physical constants
+QT = 18563.251 #For atomic tritium (eV), from Bodine et al. (2015)
+QT2 =  18573.24 #For molecular tritium (eV), Bodine et al. (2015)
+Rn =  2.8840*10**(-3) #Helium-3 nuclear radius in units of me, from Kleesiek et al. (2018): https://arxiv.org/pdf/1806.00369.pdf
+M = 5497.885 #Helium-3 mass in units of me, Kleesiek et al. (2018)
+atomic_num = 2. #For helium-3
+g =(1-(atomic_num*alpha)**2)**0.5 #Constant to be used in screening factor and Fermi function calculations
+V0 = 76. #Nuclear screening potential of orbital electron cloud of the daughter atom, from Kleesiek et al. (2018)
+mu = 5.107 #Difference between magnetic moments of helion and triton, for recoil effects correction
+
+
+"""
+Functions to calculate kinematic electron properties
+"""
+
+
+def Ee(K):
+    return K+me
+
+def pe(K):
+    return np.sqrt(K**2 + 2*K*me)
+
+def beta(K):
+    return pe(K)/Ee(K)
+
+"""
+Corrections to the simple spectrum
+"""
+
+
+#Radiative corretion to the Fermi function (from interactions with real and virtual photons)
```

Comment:
I wonder which functions defined below overlap with https://github.com/project8/mermithid/blob/master/mermithid/misc/TritiumFormFactor.py.
Depending, it might be worth merging these functions in order to avoid duplicate and confusing definitions.

Reply:
They are not duplicated. The functions here are less approximated. For example the fermi function is the relativistic one. TritiumFormFactors.py contains the non-relativistic version.